src/common.rkt

```diff
@@ -271,3 +271,10 @@
 (define (sym-append . args)
   (string->symbol (apply string-append (map ~a args))))
 
+(define/contract (string-replace* str changes)
+  (-> string? (listof (cons/c string? string?)) string?)
+  (let loop ([str str] [changes changes])
```

Comment:
Probably `for/fold` is nicer but whatever.

Reply:
Ah, I see, it's a move. Whatever then.

Reply:
A fold looks nicer, so switching to that.